test/cypress.spec.js

```diff
@@ -0,0 +1,248 @@
+const path = require('path');
+const assert = require('yeoman-assert');
+const helpers = require('yeoman-test');
+const expectedFiles = require('./utils/expected-files');
+const constants = require('../generators/generator-constants');
+
+const ANGULAR = constants.SUPPORTED_CLIENT_FRAMEWORKS.ANGULAR;
+const REACT = constants.SUPPORTED_CLIENT_FRAMEWORKS.REACT;
+const VUE = constants.SUPPORTED_CLIENT_FRAMEWORKS.VUE;
+
+describe('JHipster client generator', () => {
+    describe('generate cypress with React client with JWT', () => {
+        before(done => {
+            helpers
+                .run(path.join(__dirname, '../generators/app'))
+                .withOptions({
+                    fromCli: true,
+                    skipInstall: true,
+                    skipChecks: true,
+                })
+                .withPrompts({
+                    baseName: 'sampleMysql',
+                    packageName: 'com.mycompany.myapp',
+                    applicationType: 'monolith',
+                    databaseType: 'sql',
+                    devDatabaseType: 'h2Disk',
+                    prodDatabaseType: 'mysql',
+                    cacheProvider: 'ehcache',
+                    authenticationType: 'jwt',
+                    enableTranslation: true,
+                    nativeLanguage: 'en',
+                    languages: ['en', 'fr'],
+                    testFrameworks: ['cypress'],
+                    buildTool: 'maven',
+                    clientFramework: REACT,
+                    clientTheme: 'none',
+                })
+                .on('end', done);
```

Comment:
The following notation generates a promise, so the test is not required to timeout to fail.
```suggestion
        before(() => {
            return helpers
                .create(path.join(__dirname, '../generators/app'))
                .withOptions({
                    fromCli: true,
                    skipInstall: true,
                    skipChecks: true,
                })
                .withPrompts({
                    baseName: 'sampleMysql',
                    packageName: 'com.mycompany.myapp',
                    applicationType: 'monolith',
                    databaseType: 'sql',
                    devDatabaseType: 'h2Disk',
                    prodDatabaseType: 'mysql',
                    cacheProvider: 'ehcache',
                    authenticationType: 'jwt',
                    enableTranslation: true,
                    nativeLanguage: 'en',
                    languages: ['en', 'fr'],
                    testFrameworks: ['cypress'],
                    buildTool: 'maven',
                    clientFramework: REACT,
                    clientTheme: 'none',
                })
                .run();
```

Reply:
cc @nassimerrahoui @adilabed  ;)

Reply:
We've got this error when we try without promise :

Error: Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

Reply:
We have en error when we try with this promise notation :

`Error: Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.`

Reply:
@nassimerrahoui not sure why but it worked for me.
I refactored the test to be more modular: https://github.com/avdev4j/generator-jhipster/pull/47